include/openenclave/internal/posix/lock.h

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef _OE_POSIX_LOCK_H
+#define _OE_POSIX_LOCK_H
+
+#include <openenclave/enclave.h>
+
+#include <openenclave/corelibc/assert.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/thread.h>
+
+OE_EXTERNC_BEGIN
+
+/*
+**==============================================================================
+**
+** lock.h:
+**
+**     This file defines 'conditional lock' functions. These functions perform
+**     their operation based on the value of the locked parameter (see function
+**     descriptions below). The locked parameter must reside on the calling
+**     thread's stack. Here's a typical schenario.
```

Comment:
This requirement makes me very nervous:

- There's no way to enforce this at runtime
- It's non-obvious to infer from usage examples
- When passed down a call chain, the provenance of the variable cannot be evaluated in the context of the called function during review.

I would strongly recommend creating an actual `oe_reentrant_lock_t` with thread ID and lock count instead.

Reply:
Backed out this change completely. Thanks!